### PR TITLE
ReturnCount.excludedFunctions should be a List<String>

### PR DESCRIPTION
**Pull request by gouri-panda**
closes: #5063

### CODE REVIEW

**Comment by cortinico** [comment]
`verify-generated-config-file` is broken. Could you re-run `./gradlew generateDocumentation`?

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt [hunk 1]
```diff
@@ -59,8 +58,8 @@ class ReturnCount(config: Config = Config.empty) : Rule(config) {
     @Configuration("define the maximum number of return statements allowed per function")
     private val max: Int by config(2)
 
-    @Configuration("define a free-form comma separated list of function names to be ignored by this check")
-    private val excludedFunctions: SplitPattern by config("equals") { SplitPattern(it) }
+    @Configuration("define a list of function names to be ignored by this check")
+    private val excludedFunctions: List<String> by config(emptyList())
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:62]: You are introducing a breaking change here. You should keep `"equals"`.

And I don't know why the tests doesn't complain but looking at the code of `SplitPattern` this code doesn't do the same as before, right?

Comment (gouri-panda) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:62]: > You are introducing a breaking change here. You should keep `"equals"

Ok 
> 
> And I don't know why the tests doesn't complain but looking at the code of `SplitPattern` this code doesn't do the same as before, right?

Yes

Comment (gouri-panda) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:62]: @BraisGabin Should we change the `SplitPattern` implementation and leave the signature of `ReturnCount` as before i.e `private val excludedFunctions: SplitPattern by config("equals") { SplitPattern(it) }`?

Comment (gouri-panda) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:62]: Is this the right way to change?
`ReturnCount.kt`
```kt
    private val excludedFunctions: SplitPattern by config(listOf("equals")) { SplitPattern(it.toString()) }
```
`SplitPattern.kt`
```kt
@Suppress("detekt.SpreadOperator")
    private val excludes = text
        .commaSeparatedPattern(*delimiters.toCharArray().map { it.toString() }.toTypedArray())
        .mapIf(removeTrailingAsterisks) { seq ->
            seq.map { it.removePrefix("*") }
                .map { it.removeSuffix("*") }
                .map {it.removePrefix("[")}
                .map { it.removeSuffix("]") }
        }
        .toList()
```

Comment (BraisGabin) [reply detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:62]: I would look at what `SplitPattern` does and reproduce it in this rule. I'm sure that I did this time on other classes some time ago but for some reason I missed this one.

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt [hunk 2]
```diff
@@ -1,14 +1,8 @@
+@file:Suppress("WildcardImport", "NoWildcardImports")
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:1]: Don't use wildcard imports.

File: detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt [hunk 3]
```diff
@@ -59,8 +53,8 @@ class ReturnCount(config: Config = Config.empty) : Rule(config) {
     @Configuration("define the maximum number of return statements allowed per function")
     private val max: Int by config(2)
 
-    @Configuration("define a free-form comma separated list of function names to be ignored by this check")
-    private val excludedFunctions: SplitPattern by config("equals") { SplitPattern(it) }
+    @Configuration("define a list of function names to be ignored by this check")
+    private val excludedFunctions: SplitPattern by config(listOf("equals")) { SplitPattern(it.toString()) }
```
Comment (BraisGabin) [inline detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ReturnCount.kt:57]: If you want to go this way you can use `it.joinString(",")` (or something similar, I can't recall the signature). This way you don't need to change `SplitPattern`. That could have unexpected implications.

**Review by BraisGabin — APPROVED** [review]

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**